core/trie/proof.go

```diff
@@ -133,6 +133,163 @@ func transformNode(tri *Trie, parentKey *Key, sNode StorageNode) (*Edge, *Binary
 	return edge, binary, nil
 }
 
+func traverseNodesAndAppendToPath(path *[]ProofNode, currNode *ProofNode, nodeHashes map[string]ProofNode) error {
```

Comment:
this function & SplitPath have a lot code in common which makes me think something is way overcomplicated 🤔

Reply:
Changed this with recursive approach which traverses all paths